src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -5355,9 +5387,21 @@ static void
 finalize_dfs(void)
 {
 	int       rc, i;
+	char     *path_mt;
+	char     *path_dup;
+	char      invalid_path[] = "invalid_path";
 	d_list_t *rlink = NULL;
 
+	D_ALLOC(path_dup, DFS_MAX_PATH);
 	for (i = 0; i < num_dfs; i++) {
+		if (path_dup) {
+			path_mt = path_dup;
+			strncpy(path_mt, dfs_list[i].fs_root, dfs_list[i].len_fs_root + 1);
+		} else {
+			path_mt = invalid_path;
+		}
+		free(dfs_list[i].fs_root);
```

Comment:
Do you need any of this, can't you just call D_FREE on L5437?

Reply:
I can see what you're trying to do but please consider the following:

1. Use of DFS_MAX_PATH is technical debt which someone is going to have to remove later.  It is much easier to get it right first time than to have to come back again and re-visit this.
2. Is the error message improved by having the mount point included?  Will it affect how end-users react or avoid any issue?
3. Could the same result be achieved by simply not freeing the memory until after the other operations in this loop?